openbas-api/src/main/java/io/openbas/rest/scenario/utils/ScenarioUtils.java

```diff
@@ -21,37 +22,69 @@ private ScenarioUtils() {
   }
 
   private static final String SCENARIO_RECURRENCE_FILTER = "scenario_recurrence";
+  private static final String SCENARIO_KILL_CHAIN_PHASES_FILTER = "scenario_kill_chain_phases";
+  private static final String SCENARIO_TAGS_FILTER = "scenario_tags";
 
   /**
-   * Manage filters that are not directly managed by the generic mechanics -> scenario_kill_chain_phases
+   * Manage filters that are not directly managed by the generic mechanics -> scenario_recurrence, scenario_tags, scenario_kill_chain_phases
    */
   public static Function<Specification<Scenario>, Specification<Scenario>> handleDeepFilter(
-      @NotNull final SearchPaginationInput searchPaginationInput) {
+      @NotNull SearchPaginationInput searchPaginationInput) {
     return handleCustomFilter(searchPaginationInput);
   }
 
   private static UnaryOperator<Specification<Scenario>> handleCustomFilter(
-      @NotNull final SearchPaginationInput searchPaginationInput) {
-    // Existence of the filter
-    Optional<Filters.Filter> scenarioRecurrenceFilterOpt = ofNullable(searchPaginationInput.getFilterGroup())
-        .flatMap(f -> f.findByKey(SCENARIO_RECURRENCE_FILTER));
+      @NotNull SearchPaginationInput searchPaginationInput) {
+
+    // Extract filters from the input
+    Optional<Filters.Filter> scenarioRecurrenceFilterOpt = getFilter(searchPaginationInput, SCENARIO_RECURRENCE_FILTER);
+    Optional<Filters.Filter> scenarioKillChainsFilterOpt = getFilter(searchPaginationInput, SCENARIO_KILL_CHAIN_PHASES_FILTER);
+    Optional<Filters.Filter> scenarioTagsFilterOpt = getFilter(searchPaginationInput, SCENARIO_TAGS_FILTER);
```

Comment:
Do you think we can have a generic system to remove double join ?
Check if the join is already there and don't do it again ?


I think we have this kind of problem on exercise and teams too

Reply:
Yes, I think it’s better to find another generic solution, but the double joins are normal and are indeed created based on our requests. For example, if we ask for two different attributes from 'inject,' the database has to perform double joins with 'inject.' In this case (scenarios), the query also has a raw format, which requires including the joins for tags and platform in the query base.

Reply:
We have a join from the toPath method in the resolution of filter
And we have a join in the criteriabuilder when we build the scenario object

The way you propose can be a good solution if we can't arrive to generalize something. 

We can try to reuse the join from the scenario object when we resolve filters. Maybe by having a request context to populate with a map of Join object.

Reply:
Yeah, I initially tried to use the same join inject.contract for both platforms and kill chains (with modified paths and move getters) but I couldn't get it to work. I’m going to re-try the two options you suggested: 1. Verify duplicated joins, and 2. Force the use of the same joins. Thanks!